Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -32,36 +36,50 @@ noncomputable section
 
 open Function Set Subalgebra MvPolynomial Algebra
 
-universe u v w
+universe u u' v w
 
-variable {ι ι' : Type*} (R : Type*) {K A A' : Type*}
-variable {x : ι → A}
-variable [CommRing R] [CommRing A] [CommRing A'] [Algebra R A] [Algebra R A']
+variable {ι : Type u} {ι' : Type u'} (R : Type*) {S : Type v} {A : Type w}
+variable {x : ι → A} {y : ι' → A}
+variable [CommRing R] [CommRing S] [CommRing A]
+variable [Algebra R S] [Algebra R A] [Algebra S A] [IsScalarTower R S A]
 
 open AlgebraicIndependent
 
-theorem exists_isTranscendenceBasis (h : Injective (algebraMap R A)) :
+variable {R} in
+theorem exists_isTranscendenceBasis_superset {s : Set A}
+    (hs : AlgebraicIndepOn R id s) :
+    ∃ t, s ⊆ t ∧ IsTranscendenceBasis R ((↑) : t → A) := by
+  simpa only [subset_univ, and_true, ← isTranscendenceBasis_iff_maximal]
+    using exists_maximal_algebraicIndependent s _ (subset_univ _) hs
+
+variable (A)
+theorem exists_isTranscendenceBasis [FaithfulSMul R A] :
     ∃ s : Set A, IsTranscendenceBasis R ((↑) : s → A) := by
-  obtain ⟨s, hs⟩ := exists_maximal_algebraicIndependent (∅ : Set A) Set.univ (Set.subset_univ _)
-      ((algebraicIndependent_empty_iff R A).2 h)
-  refine ⟨s, hs.2.1.1, fun t ht hst ↦ ?_⟩
-  simp only [Subtype.range_coe_subtype, setOf_mem_eq] at *
-  exact hs.2.eq_of_le ⟨ht, subset_univ _⟩ hst
+  simpa only [empty_subset, true_and] using exists_isTranscendenceBasis_superset
```

Comment:
Same.

Reply:
I'm not sure if we should unsqueeze simp calls with two lemmas.

Reply:
Yeah, and these are pretty basic lemmas that are unlikely to be changed, so they won't become a maintenance burden.

Reply:
```suggestion
  simpa using exists_isTranscendenceBasis_superset
```

Reply:
Okay, unsqueezed!